File: backend/src/AqlanDentalPro.Domain/Entities/Employee.cs
namespace AqlanDentalPro.Domain.Entities;

public class Employee : BaseEntity
{
    public Guid UserId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? NationalId { get; set; }
    public string? Position { get; set; }
    public Guid? BranchId { get; set; }
    public DateTime? HireDate { get; set; }
    public decimal? BaseSalary { get; set; }
    public string? EmergencyContact { get; set; }
    public string? EmergencyPhone { get; set; }
    public string? Notes { get; set; }

    public User User { get; set; } = null!;
    public Branch? Branch { get; set; }
}

File: .migration-backup/frontend/src/types/employee.ts
export interface Employee {
  id: string;
  userId: string;
  fullName: string;
  phone?: string;
  nationalId?: string;
  position?: string;
  branchId?: string;
  branchName?: string;
  hireDate?: string;
  baseSalary?: number;
  emergencyContact?: string;
  emergencyPhone?: string;
  notes?: string;
  username: string;
  role: string;
  userActive: boolean;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateEmployeeRequest {
  fullName: string;
  phone?: string;
  nationalId?: string;
  position?: string;
  branchId?: string;
  hireDate?: string;
  baseSalary?: number;
  emergencyContact?: string;
  emergencyPhone?: string;
  notes?: string;
  username: string;
  password?: string;
  role: string;
}

export interface UpdateEmployeeRequest {
  fullName?: string;
  phone?: string;
  nationalId?: string;
  position?: string;
  branchId?: string;
  hireDate?: string;
  baseSalary?: number;
  emergencyContact?: string;
  emergencyPhone?: string;
  notes?: string;
}

export const ROLE_LABELS: Record<string, string> = {
  Admin: "مدير النظام",
  Orthodontist: "أخصائي تقويم",
  GeneralDentist: "طبيب أسنان",
  OralSurgeon: "جراح وجه وفكين",
  Reception: "استقبال",
  Accountant: "محاسب",
  Assistant: "مساعد",
  BranchManager: "مدير فرع",
};

export const POSITION_LABELS: Record<string, string> = {
  doctor: "طبيب",
  reception: "موظف استقبال",
  assistant: "مساعد طبيب",
  nurse: "ممرض/ة",
  accountant: "محاسب",
  manager: "مدير",
  technician: "فني أسنان",
  cleaner: "عامل نظافة",
  other: "أخرى",
};
